watertap/core/wt_database.py

```diff
@@ -0,0 +1,111 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+This module contains the base class for interacting with WaterTAP data files
+with zero-order model parameter data.
+"""
+import os
+import yaml
+
+
+class Database:
+    def __init__(self, dbpath=None):
+        self._cached_files = {}
+
+        if dbpath is None:
+            self._dbpath = os.path.join("watertap", "data", "techno_economic")
+        else:
+            self._dbpath = dbpath
+
+            # Confirm valid path
+            if not os.path.isdir(self._dbpath):
+                raise OSError(
+                    f"Could not find requested path {self._dbpath}. Please "
+                    f"check that this path exists.")
+
+    def get_unit_operation_parameters(self, technology, subtype=None):
+        """
+        Method to retrieve parameters for a given technology by subtype.
+
+        Args:
+            technology - unit operation technology to look up and retrieve
+                         parameters for.
+            subtype - (optional) string or list-of-strings indicating specific
+                      sub-type of technology to return parameters for. If not
+                      provided, the default parameters are used instead.
```

Comment:
I commented on this above, but do we necessarily want the default parameters to be used instead?

Reply:
Two initial ways I see to do this are 
1) in the data file, include redundant keys under subtype which are assigned a value of NaN when data isn't relevant or available for that particular parameter
2) Add an optional argument which would allow either a) use of default parameters for missing parameters (as we have it now) or b) replace those missing parameters with NaNs (or remove them entirely?)

Reply:
If the user just want to change one or two things, forcing them to specify every other parameter is cumbersome. This is a perhaps a bit more dangerous, but is still explicit. The current implementation is a pattern I've seen used by other tools: it can be cumbersome if you layer too many sub-types on top of each other, but that's an anti-pattern we guard against.